
# GO å¯Œé›†ç»“æœ_ggplot2 ç»˜å›¾   

```r
library(ggplot2)
MSU_160_GO<-read.csv("C:\\Users\\zhuwengen\\Desktop\\tangyimiao\\GO\\MSU_160_GO.csv",stringsAsFactors = TRUE)
colnames(MSU_160_GO)      # ç»˜å›¾ä¸»è¦ä½¿ç”¨"Rank"ï¼Œ"Description" ï¼Œ"Count" åˆ—
[1] "Rank"             "ID"               "Description"      "GeneRatio"        "BgRatio"         
[6] "pvalue"           "p.adjust"         "qvalue"           "geneID"           "Count"           
[11] "percent_of_genes"
```

MSU_160_GO æ•°æ®ç»“æ„,å¦‚ä¸‹å›¾ 
  
![](../picture/MSU_160_GO.png)      

## åˆ†é¢ï¼ç»˜å›¾  
```
ggplot(data = MSU_160_GO, mapping = aes(reorder(Description,Count),Count, fill=Rank)) +
        geom_bar(aes(fill=Rank),stat= 'identity', width = 0.5) +
        facet_grid(~Rank,space="free_x", scales="free_x") + 
         #æ¢ä¸ªè§’åº¦æ˜¾ç¤ºï¼Œangleæ˜¯æ§åˆ¶åˆ»åº¦æ ‡ç­¾æ˜¾ç¤ºçš„è§’åº¦ + 
        theme(axis.text.x = element_text(angle=30, hjust=1, vjust=1))
```
 
![](../picture/GO_160_BP_CC_MF.pdf)      





ç”¨ Abraham Lincoln çš„è¯æ¥è¯´ï¼š

> åŸè°…æˆ‘çˆ†ç²—å£

### ä¸‰åŸºå› ç»„å…±çº¿æ€§åˆ†æ
æœ¬æ•™ç¨‹å‚è€ƒï¼š[MCscan (Python version)å®˜ç½‘](https://genehub.wordpress.com/2019/07/05/mcscanx-%E6%B5%8B%E8%AF%95/)ï¼Œæœ‰ä¿®æ”¹ï¼

#### ç¬¬ä¸€æ­¥ï¼šå‡†å¤‡åˆ†æç¯å¢ƒ  
```
$ conda create -n jcvi python=2.7   # å¦‚å·²åˆ›å»ºè¯¥ç¯å¢ƒï¼Œè¯·å¿½ç•¥è¿™æ­¥
$ conda install jcvi last scipy     # å¦‚å·²ç»å®‰è£…è¿™äº›æ¨¡å—ï¼Œè¯·å¿½ç•¥è¿™æ­¥

$ conda activate jcvi                               # æ¿€æ´»å·²ç»å®‰è£…jcviæ¨¡å—çš„jcviç¯å¢ƒ
$ mkdir Grape.Peach.Cacao  | cd Grape.Peach.Cacao   # åœ¨mcscan_between_genomeç›®å½•ä¸‹åˆ›å»º
```

#### ç¬¬äºŒæ­¥ï¼šä¸‹è½½ã€è§£å‹æ¡ƒğŸ‘ã€è‘¡è„ğŸ‡ã€å¯å¯ğŸ¥¥çš„ gffå’Œcds æ•°æ®  
ä¸‹è½½ï¼š Prunus persica (peach)ã€Vitis vinifera L.(grape)ã€Theobroma cacao(cacao)çš„cds å’Œ gffæ–‡ä»¶

```
$ ls
Ppersica_298_v2.1.cds.fa.gz     Tcacao_523_v2.1.cds.fa.gz     Vvinifera_145_Genoscope.12X.cds.fa.gz
Ppersica_298_v2.1.gene.gff3.gz  Tcacao_523_v2.1.gene.gff3.gz  Vvinifera_145_Genoscope.12X.gene.gff3.gz

$ gunzip *gz

$ ll
-rw-rw-r-- 1 manager manager 71828035 May  5 12:58 Ppersica_298_v2.1.cds.fa
-rw-rw-r-- 1 manager manager 61401929 May  5 12:58 Ppersica_298_v2.1.gene.gff3
-rw-rw-r-- 1 manager manager 58569425 May  5 12:57 Tcacao_523_v2.1.cds.fa
-rw-rw-r-- 1 manager manager 54953955 May  5 12:57 Tcacao_523_v2.1.gene.gff3
-rw-rw-r-- 1 manager manager 34297711 May  5 12:57 Vvinifera_145_Genoscope.12X.cds.fa
-rw-rw-r-- 1 manager manager 34622840 May  5 12:57 Vvinifera_145_Genoscope.12X.gene.gff3
```


#### ç¬¬ä¸‰æ­¥ï¼šå°†gffè½¬æˆbedæ ¼å¼  
**æ³¨æ„**ï¼šæœ¬æ–‡çš„ç‰©ç§å‚æ•°è®¾ç½®ä¸º `--key=Name`ï¼Œå³é€‰è½¬å½•æœ¬idï¼›æœ‰äº›ç‰©ç§é€‰`--key=ID`ï¼Œå…·ä½“éœ€è¦å…ˆæŸ¥çœ‹gffæ–‡ä»¶ã€‚
```
$ python -m jcvi.formats.gff bed --type=mRNA --key=Name Tcacao_523_v2.1.gene.gff3 -o cacao.bed
$ python -m jcvi.formats.gff bed --type=mRNA --key=Name Ppersica_298_v2.1.gene.gff3 -o peach.bed
$ python -m jcvi.formats.gff bed --type=mRNA --key=Name Vvinifera_145_Genoscope.12X.gene.gff3 -o grape.bed  

# å¯ä¹Ÿå¯¹bedæ–‡ä»¶ä¸­çš„åŸºå› è¿›è¡Œå»é‡(å¯ä¸ç”¨æ­¤æ­¥éª¤)  
$ python -m jcvi.formats.bed uniq cacao.bed  
$ python -m jcvi.formats.bed uniq peach.bed  
$ python -m jcvi.formats.bed uniq grape.bed  
```  

#### ç¬¬å››æ­¥ï¼šå‡†å¤‡CDSåºåˆ—   
```
$ python -m jcvi.formats.fasta format --sep="|" Tcacao_523_v2.1.cds.fa cacao.cds 
$ python -m jcvi.formats.fasta format --sep="|" Ppersica_298_v2.1.cds.fa peach.cds 
$ python -m jcvi.formats.fasta format --sep="|" Vvinifera_145_Genoscope.12X.cds.fa grape.cds 

# è¿™ä¸€æ­¥å¾ˆé‡è¦ï¼ï¼ï¼éœ€è¦å…ˆæŸ¥çœ‹cdsåºåˆ—ï¼Œç„¶åç”¨æ­£åˆ™è¡¨è¾¾å¼æ›¿æ¢ï¼   
$ sed  -i 's#pacid.*##' cacao.cds  
$ sed  -i 's#pacid.*##' peach.cds  
$ sed  -i 's#pacid.*##' grape.cds    
```  

#### ç¬¬äº”æ­¥ï¼šæ¡ƒå­(peach)å’Œè‘¡è„(grape)å…±çº¿æ€§åˆ†æï¼›è‘¡è„(grape)å’Œå¯å¯(cacao)å…±çº¿æ€§åˆ†æ
```
$ python -m jcvi.compara.catalog ortholog peach grape --cscore=0.7 --no_strip_names &>stdout
$ python -m jcvi.compara.catalog ortholog grape cacao --cscore=0.7 --no_strip_names &>stdout

# æŸ¥çœ‹éƒ¨åˆ†ç»“æœæ–‡ä»¶(peach~grape )
-rw-r--r-- 1 root    root     29984686 May 19 15:32 grape.tis
-rw-r--r-- 1 root    root       105388 May 19 15:32 grape.ssp
-rw-r--r-- 1 root    root       105388 May 19 15:32 grape.sds
-rw-r--r-- 1 root    root       474228 May 19 15:32 grape.des
-rw-r--r-- 1 root    root    115035544 May 19 15:32 grape.suf
-rw-r--r-- 1 root    root          497 May 19 15:32 grape.prj
-rw-r--r-- 1 root    root     24566708 May 19 15:32 grape.bck
-rw-r--r-- 1 root    root     57882088 May 19 15:32 peach.grape.last
-rw-r--r-- 1 root    root      1739132 May 19 15:32 peach.grape.last.filtered
-rw-r--r-- 1 root    root       565490 May 19 15:32 peach.grape.anchors
-rw-r--r-- 1 root    root      1922859 May 19 15:32 peach.grape.lifted.anchors
drwxrwxrwx 2 root    root         4096 May 19 15:32 ./
-rw-r--r-- 1 root    root         5062 May 19 15:33 stdout
-rw-r--r-- 1 root    root        91197 May 19 15:33 peach.grape.pdf
```

#### ç¬¬å…­æ­¥ï¼šç»˜å›¾æ–‡ä»¶å‡†å¤‡(ğŸ˜­)
1. `.simple`: ä».anchorsæ–‡ä»¶åˆ›å»ºçš„æ›´ç®€åŒ–æ ¼å¼
```
$ python -m jcvi.compara.synteny screen --minspan=30 --simple peach.grape.anchors peach.grape.anchors.new 
$ python -m jcvi.compara.synteny screen --minspan=30 --simple grape.cacao.anchors grape.cacao.anchors.new
```

2. `layout`: ä¸åŒç‰©ç§åœ¨å›¾ä¸Šçš„ä½ç½®  
**æ³¨æ„**ï¼špeach.grape.anchors.simple é¡ºåº!!!,peach åœ¨å‰ï¼Œgrapeåœ¨åã€‚  
```  
$ vi layout  
# y, xstart, xend, rotation, color, label, va,  bed
 .7,     .1,    .8,      15,      , Peach, top, peach.bed
 .5,     .1,    .8,       0,      , Grape, top, grape.bed
 .3,     .1,    .8,     -15,      , Cacao, bottom, cacao.bed
# edges
e, 0, 1, peach.grape.anchors.simple
e, 1, 2, grape.cacao.anchors.simple
```  

3. `seqids`: éœ€è¦å±•ç°å“ªäº›åºåˆ—  
**æ³¨æ„**ï¼šè¦æ ¹æ®bedæ–‡ä»¶çš„æŸ“è‰²ä½“åå­—æ¥å¡«å†™!!!  
è¯¥æ–‡ä»¶è§„å®šå›¾ä¸­å„åŸºå› ç»„çš„æŸ“è‰²ä½“é¡ºåºå’Œéœ€è¦æ˜¾ç¤ºçš„æŸ“è‰²ä½“ï¼Œä¸€è¡Œä»£è¡¨ä¸€ä¸ªåŸºå› ç»„ï¼Œå…¶é¡ºåºè¦ä¸layoutä¸­åŸºå› ç»„é¡ºåºä¸€è‡´ã€‚
```
$ vi seqids
Pp01,Pp02,Pp03,Pp04,Pp05,Pp06
chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19
Chromosome_1,Chromosome_2,Chromosome_3,Chromosome_4,Chromosome_5
```

#### ç¬¬ä¸ƒæ­¥ï¼šç»˜å›¾  
```
$ python -m jcvi.graphics.karyotype seqids layout 
```


å›¾ç‰‡å¦‚ä¸‹    
![](./picture/Peach_Grape_Cacao_karyotype.jpg)  



å›¾ç‰‡å¦‚ä¸‹   
![](./picture/grape_peach_synteny_depth.png)







































































